desktop/apps/consignments2/components/location_autocomplete/index.js

```diff
@@ -12,56 +12,75 @@ import {
   updateLocationAutocomplete
 } from '../../client/actions'
 
-function getSuggestionValue (suggestion) {
-  return suggestion.description
-}
-
-function renderSuggestion (suggestion) {
-  return (
-    <div className='autosuggest-suggestion'>
-      <div>{suggestion.description}</div>
-    </div>
-  )
-}
-
 function LocationAutocomplete (props) {
   const {
     chooseLocationAction,
     clearLocationSuggestionsAction,
     fetchLocationSuggestionsAction,
+    input,
     locationAutocompleteFrozen,
     locationAutocompleteSuggestions,
     locationAutocompleteValue,
+    meta: {
+      error,
+      touched
+    },
     unfreezeLocationInputAction,
     updateLocationAutocompleteAction
   } = props
 
   const b = block('consignments2-location-input')
 
   const locationAutosuggestInputProps = {
-    value: locationAutocompleteValue,
-    onChange: updateLocationAutocompleteAction
+    disabled: locationAutocompleteFrozen,
+    onChange: updateLocationAutocompleteAction,
```

Comment:
In order to make the validation simple, I added the `location` autocomplete to the redux-form. Since we still have very specific handling of that element/values, I think it makes sense to not handle it _purely_ within redux-form.

Right now, we update the `'location'` value in our form to be either the _full_ selection or `''` based on when they `select` something or unfreeze the box.